web/src/pages/user.jsx

```diff
@@ -18,10 +18,39 @@ export const SongMetadata = {
     albumArtUrl: undefined
 };
 
-export default ({songs}) => {
+export default ({songsProp}) => {
     const [name, setName] = useState("Loading...");
     const [searchedSongs, setSearchedSongs] = useState([]);
     const [searchQuery, setSearchQuery] = useState("");
+    const [loading, setLoading] = useState(false);
+    const [songs, setSongs] = useState(songsProp);
+    
+
+    const fetchMoreSongs = async (searchString) => {
+        try {
+            setLoading(true);
+            const response = await fetch("/api/spotify/search", {
+                method: "POST",
+                headers: {
+                    "Content-Type": "application/json"
+                },
+                body: JSON.stringify({search_string: searchString})
+            });
+
+            if (response.ok) {
+                const data = await response.json();
+                setSongs([...songs, ...data]);
```

Comment:
This doesn't work, if you keep searching up the same string it will continue appending duplicate entries. There needs to be some sort of filter (likely by id or by some key) to prevent duplicates.

<img width="1324" alt="image" src="https://github.com/timrolsh/Songscope/assets/68395788/3e3e1d04-b8a3-4e65-8fc2-9dc239c8291e">

Issue is shown in the image above.

Reply:
Doesn't look like duplicates are there anymore